### PR TITLE
chore: Reduce enzymes usage 1/3

### LINKED ISSUE #8616: Reduce enzyme usage in unit test by 25% - [1 of 3]
### What is this about?

Convert the list of unit tests to use `react-testing-library` instead of `enzyme`. In total there are 53 files to convert. To make PRs easier to review and more manageable, this covers 17 test files.

### Scenario

_No response_

### Design

_No response_

### Technical Details

Convert these files

```
'app/components/Views/Root/index.test.tsx',
'app/components/Views/NavigationUnitTest/TestScreen3.test.js',
'app/components/Views/NavigationUnitTest/TestScreen2.test.js',
'app/components/Views/NavigationUnitTest/TestScreen1.test.js',
'app/components/Views/ImportPrivateKeySuccess/index.test.tsx',
'app/components/Views/AccountBackupStep1B/index.test.tsx',
'app/components/UI/WebviewProgressBar/index.test.tsx',
'app/components/UI/SeedphraseModal/index.test.tsx',
'app/components/UI/PhishingModal/index.test.tsx',
'app/components/UI/FoxScreen/index.test.tsx',
'app/components/UI/FadeView/index.test.tsx',
'app/components/UI/CustomAlert/index.test.tsx',
'app/components/UI/Confetti/index.test.tsx',
'app/components/UI/Collectibles/index.test.tsx',
'app/components/UI/Button/index.test.tsx',
'app/components/UI/AnimatedSpinner/index.test.tsx',
'app/components/UI/AddCustomToken/index.test.tsx',
```

### Threat Modeling Framework

_No response_

### Acceptance Criteria

Unit tests for the updated files should all pass. Snapshots may be updated

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by timi-codes**
## **Description**

Convert the list of unit tests to use react-testing-library instead of enzyme for the following files:
```
'app/components/Views/Root/index.test.tsx',
'app/components/Views/NavigationUnitTest/TestScreen3.test.js',
'app/components/Views/NavigationUnitTest/TestScreen2.test.js',
'app/components/Views/NavigationUnitTest/TestScreen1.test.js',
'app/components/Views/ImportPrivateKeySuccess/index.test.tsx',
'app/components/Views/AccountBackupStep1B/index.test.tsx',
'app/components/UI/WebviewProgressBar/index.test.tsx',
'app/components/UI/SeedphraseModal/index.test.tsx',
'app/components/UI/PhishingModal/index.test.tsx',
'app/components/UI/FoxScreen/index.test.tsx',
'app/components/UI/FadeView/index.test.tsx',
'app/components/UI/CustomAlert/index.test.tsx',
'app/components/UI/Confetti/index.test.tsx',
'app/components/UI/Collectibles/index.test.tsx',
'app/components/UI/Button/index.test.tsx',
'app/components/UI/AnimatedSpinner/index.test.tsx',
'app/components/UI/AddCustomToken/index.test.tsx',
```

## **Related issues**

Fixes: #8616 

## **Manual testing steps**

Not Applicable

## **Screenshots/Recordings**

Not Applicable

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Cal-L — COMMENTED** [review]
Left comments of simplified tests using existing utility functions

**Comment by Cal-L** [comment]
@timi-codes Left some comments

**Comment by timi-codes** [comment]
Thanks for the feedback @Cal-L 

I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx`  and `app/components/UI/Confetti/index.test.tsx`.

The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. 
Do you have any suggestion on how to fix this?

image
image

**Comment by Cal-L** [comment]
> Thanks for the feedback @Cal-L
> 
> I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx` and `app/components/UI/Confetti/index.test.tsx`.
> 
> The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. Do you have any suggestion on how to fix this?
> 
> image image

On `Confetti.test.tsx`, since the components being tested have non deterministic styles, I'd recommend just mocking `react-native-confetti` and `react-native-confetti-cannon` for now:

```
jest.mock('react-native-confetti');
jest.mock('react-native-confetti-cannon');
```

The snapshot will reduce to only return null but that's fine for now since this isn't a critical component.

**Comment by Cal-L** [comment]
> Thanks for the feedback @Cal-L
> 
> I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx` and `app/components/UI/Confetti/index.test.tsx`.
> 
> The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. Do you have any suggestion on how to fix this?
> 
> image image

For the Root component snapshot tests, you can update the test file to look like this:

image

After that, you can remove some of the mocks

image

Your Root snapshot should update and the output should remain deterministic

**Comment by Cal-L** [comment]
@timi-codes Left some comments regarding Confetti and Root components

**Comment by timi-codes** [comment]
> > Thanks for the feedback @Cal-L
> > I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx` and `app/components/UI/Confetti/index.test.tsx`.
> > The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. Do you have any suggestion on how to fix this?
> > image image
> 
> For the Root component snapshot tests, you can update the test file to look like this:
> 
> image
> After that, you can remove some of the mocks
> 
> image
> Your Root snapshot should update and the output should remain deterministic

@Cal-L Thank you. 

For the Root file, I was referring to `app/components/Views/Root/index.test.tsx` not `app/components/Views/confirmations/components/SignatureRequest/Root/Root.test.tsx`. However, I already fixed it by mocking `configureStore`. 

All the files are now ready to be reviewed.

File: app/components/UI/Button/index.test.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render } from '@testing-library/react-native';
 import Button from './';
 
 describe('Button', () => {
   it('should render correctly', () => {
-    const wrapper = shallow(<Button />);
+    const wrapper = render(<Button />);
```
Comment (Cal-L) [inline app/components/UI/Button/index.test.tsx:7]: check for isJSON

File: app/components/UI/FadeView/index.test.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render } from '@testing-library/react-native';
 import FadeView from './';
 
 describe('FadeView', () => {
   it('should render correctly', () => {
-    const wrapper = shallow(<FadeView visible />);
+    const wrapper = render(<FadeView visible />);
```
Comment (Cal-L) [inline app/components/UI/FadeView/index.test.tsx:7]: check for isJSON

**Review by Cal-L — COMMENTED** [review]
Left some comments @timi-codes

**Review by Cal-L — APPROVED** [review]
LGTM, thanks for your contribution!

**Comment by timi-codes** [comment]
Thank you @Cal-L . The 2/3 is also ready for review https://github.com/MetaMask/metamask-mobile/pull/8823

**Comment by Cal-L** [comment]
@timi-codes  Merging into a branch based on main and will follow up with a PR to merge into main.

**Pull request merged**